### ISSUE TITLE
[Question] Setting up relation layer without label to its span

### ISSUE THREAD

**Issue by ofersabo** [inception-project/inception#1958]
Hi all, 

in our project, we have two layers, a "Span" layer, and a "Relation" layer.
The span layer is designed for annotators to mark spans that correspond to an event.
as an example, "pain" is considered as an event. 

The Relation layer assigns features to the event from the first layer.
See the attached image as an example: 
![Screen Shot 2021-01-23 at 16 40 25](https://user-images.githubusercontent.com/31179968/105589365-6ffb0280-5d9a-11eb-934d-ec1216d21f88.png)

The span "injury" is an event and the "chest" span is a feature of that injury event. 
This particular feature describes the location of the injury. 
As admins, we assign a few feature options for the relation layer.
To illustrate, the features could be one of these options: 
"location", "type of injury", "intensity of pain".

Assuming an annotator connects a feature to an event, we want to force the user to choose a label (out of the three possible "location", "type of injury", "intensity of pain")  to the connection, i.e. not to leave the connection label blank.
The attached image describes the scenario which we want to avoid
![image](https://user-images.githubusercontent.com/31179968/105598171-d7b24d00-5d9c-11eb-9f13-b08b67faaa21.png)
 
To this end, in setting under the layer tab, we checked the box that says "required" near to where we assign tagset to the relation layer.
see image. 
![image](https://user-images.githubusercontent.com/31179968/105591855-20690680-5d9b-11eb-828d-e7774202affe.png)

However, this action also requires the annotator to assign a label to the span.
in our example, the user is required to set a label to the span "chect"
This is not the behavior we want. we don't need any label on the feature spans, just the connection label. 

To summarize, our goal is to force the user to put a label only on the connection, on the edge itself and not on the span.

Can this be done?  
Did we configure the project correctly? using the "Span" and "Relation" layers?

**Comment by reckart** [comment]
I need to try and reproduce the potentially odd behavior of the "require" option - there may be a bug.

Regarding modelling event annotation: you can do it using spans and relations. However, we did develop the "slot feature" specifically for event annotation. You can e.g. take a look at the built-in "SemPred" and "SemArg" layers. I believe that modelling events using the approach of using a slot feature as done in SemPred/SemArg is more flexible and better suited  - but you may have requirements causing you do find the approach inferior to a span/relation approach. 

I am afraid we currently do not have an example project showcasing event annotation, but you can see a cropped screenshot of an example [here](https://inception-project.github.io/use-cases/rapid-events/).

**Comment by reckart** [comment]
So I have created a new project, a span layer without a feature and a relation layer with a string feature called label. I have marked the label feature to be required. For me, only the relation is marked red on the annotation page if I do not assign a label.

<img width="329" alt="Screenshot 2021-01-23 at 17 13 46" src="https://user-images.githubusercontent.com/1410238/105607386-a175cd00-5d9e-11eb-967f-588d99e789b2.png">

Please check that only the relation feature is marked as required and not the feature on the span layer.

**Comment by ofersabo** [comment]
I believe the issue is that both the span University and Chicago are considered part of the Span layer.
Although the university span should not be of the same layer as the Chicago span. 
In case they share the same layer, it means that if the Span is also required then the span University will be also forced into some label.
This is the case on my side.

**Comment by reckart** [comment]
It is a limitation of the current implementation that relations can only attach to one span layer - i.e. you cannot have relations between different types of layers. That limitation does not exist for slot features though. You can choose a slot feature either to attach to a particular type of span layer **or** you can choose it to attach to **any** type of spans. That allows you to have an event where the "features" of the event can be from different other layers.

Btw. here is another example - frame annotation using a slot feature:

<img width="1217" alt="Screenshot 2021-01-23 at 17 25 03" src="https://user-images.githubusercontent.com/1410238/105607668-6d9ba700-5da0-11eb-8ad9-d29dfb3ac663.png">

You can also combine the slot feature with "constraints". E.g. you can write a rule saying that events of a certain type typically have a "location" or "actor" slot and these slots can then be pre-created in the UI so that the annotators only have to fill them. I'll try to pack an example together and upload that to our website...

**Comment by ofersabo** [comment]
Thanks @reckart 

 "SemPred" and "SemArg" layers do seem a better fit to event extraction, I guess the same applies for SRL.
Before I move my project from span and relation layer to the suggested layers I'd do more testing to see what I may miss if we change the project setting. The change requires moving already annotated documents from the older setting to the new settings.

**Comment by reckart** [comment]
Btw. I added a basic example for frame annotation here: https://inception-project.github.io/example-projects/frame-annotation/

**Comment by ofersabo** [comment]
Thanks, just trying this projcet at the moment, can you elaborate on the constraints you put in this example project?

Particualrllly, I'm interested at these constraints:

  category = "Questioning" ->
    arguments.role = "Addressee" (!) |
    arguments.role = "Message" (!) |
    arguments.role = "Speaker" (!) |
    arguments.role = "Time" |
    arguments.role = "Iterations";

**Comment by reckart** [comment]
Does the [manual](https://inception-project.github.io/releases/0.17.4/docs/user-guide.html#_constraints_for_slot_features) help?

**Comment by ofersabo** [comment]
Yes, Thank you!
